Test `load-data` ns

Eleonore9 opened the pull request:
- Add tests for data loading & schema generation in `load-data` namespace
- Spaces, /, apostrophes in input data column names get replaced with '-'
- Add description & circle CI badge to README

Commit:
Add tests for the load-data ns

Add circleCI badge + description

Test for schema generation functions

otfrom commented on test/witan/models/load_data_test.clj:
```diff
@@ -0,0 +1,67 @@
+            [schema.core :as s]))
+
+(defn- same-coll? [coll1 coll2]
+  (= (set coll1) (set coll2)))
```
are you sure about using set here?

```
user> (= (set [1 1 2]) (set [1 2]))
true
user> (= [1 1 2] [1 2])
false
```

Eleonore9 replied:
I think we need to return as equal two collections with the same content but possibly of different type (vector or list) and ordered differently.

otfrom replied:
```
user> (= '(1 1 2) [1 1 2])
true
```

and won't order be significant? 

if you do want it w/o significant order then I'd do:

```
user> (= (sort '(1 1 2)) (sort [1 2 1]))
true
```

but converting to sets gets rid of duplicate values, which may or may not be what you want.

otfrom commented on test/witan/models/load_data_test.clj:
```diff
@@ -0,0 +1,67 @@
+(deftest custom-keyword-test
+  (testing "String column names are keywordised, with '.' and ' ' replaced by '-'."
+    (is (= [:col-1 :col2 :col-3]
+           (#'witan.models.load-data/custom-keyword ["col 1" "col2" "col.3"])))))
```
testing private functions always makes me itchy. If they are private it is b/c they are part of the implementation and might change w/o changing the interface. I'd expect to be able to change a private function without having to change a test. Same with the use of `load-csv` below.

Eleonore9 replied:
Np, these functions don't need to be private.

LoraArmstrong committed:
Changes from @otfrom's comments

Remove tests for private functions
Make separate fn for loading single dataset
Account for invalid characters in column names

Test for load-dataset, remove unneeded tests

Formatting

LoraArmstrong committed:
Update test after rebase

otfrom commented on test/witan/models/load_data_test.clj:
```diff
@@ -0,0 +1,55 @@
+    (testing "Loaded dataset has correct column names"
+      (is (= [:gss-code :sex :age :births :year] (ds/column-names data))))
+    (testing "Dataset was loaded"
+      (is (= nil (nils-in-map? data-map))))
```
`(is (every? identity (vals m)))` instead? I _think_ you are saying there shouldn't be any nil values in the map.

LoraArmstrong replied:
thanks. yes, that was the goal.

otfrom commented on test/witan/models/load_data_test.clj:
```diff
@@ -0,0 +1,55 @@
+(defn nils-in-map?
+  "Returns true if any values are nil, otherwise returns nil"
+  [m]
+  (some true? (map nil? (vals m))))
```
this function probably isn't needed given `every?` below.

LoraArmstrong committed:
Changes from @otfrom's comments, cljfmt fix

otfrom commented:
:shipit:
